Warcraft.NET/Files/ADT/TerrainLOD/Legion/TerrainLOD.cs

```diff
@@ -0,0 +1,164 @@
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        [ChunkOrder(2)]
+        public MLHD Header { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [ChunkOrder(3)]
+        public MLVH Heightmap { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [ChunkOrder(4)]
+        public MLLL Levels { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [ChunkOrder(5)]
+        public MLND QuadTree { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [ChunkOrder(6)]
+        public MLVI VertexIndices { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [ChunkOrder(7)]
+        public MLSI SkirtIndices { get; set; }
+
+        /// <summary>
+        /// Blend mesh headers.
+        /// </summary>
+        [ChunkOrder(8), ChunkOptional]
+        public MBMH BlendMeshHeaders { get; set; }
+
+        /// <summary>
+        /// Blend mesh bounding boxes.
+        /// </summary>
+        [ChunkOrder(9), ChunkOptional]
+        public MBBB BlendMeshBoundingBoxes { get; set; }
+
+        /// <summary>
+        /// Blend mesh indices.
+        /// </summary>
+        [ChunkOrder(10), ChunkOptional]
+        public MBMI BlendMeshIndices { get; set; }
+
+        /// <summary>
+        /// Blend mesh vertices.
+        /// </summary>
+        [ChunkOrder(11), ChunkOptional]
+        public MBNV BlendMeshVertices { get; set; }
+
+        /// <summary>
+        /// Blend mesh batches.
+        /// </summary>
+        [ChunkOrder(12), ChunkOptional]
+        public MBMB BlendMeshBatches { get; set; }
+
+        /// <summary>
+        /// Liquid data.
+        /// TODO: Requires additional implementation for decompressing as well as properly reading the last chunk.
+        /// </summary>
+        [ChunkOrder(13), ChunkOptional]
+        public MLLD LiquidData { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public SynchronizedList<MLLN?> LiquidN { get; set; } = new SynchronizedList<MLLN?>(new List<MLLN?>(4096));
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public SynchronizedList<MLLI?> LiquidIndices { get; set; } = new SynchronizedList<MLLI?>(new List<MLLI?>(4096));
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public SynchronizedList<MLLV?> LiquidVertices { get; set; } = new SynchronizedList<MLLV?>(new List<MLLV?>(4096));
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TerrainLOD"/> class.
+        /// </summary>
+        public TerrainLOD() : base()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TerrainLOD"/> class.
+        /// </summary>
+        /// <param name="inData">The binary data.</param>
+        public TerrainLOD(byte[] inData)
+        {
+            using var ms = new MemoryStream(inData);
```

Comment:
The read of the data is mostly done in a `public void LoadBinaryData(byte[] inData)` function which get called by the `public TerrainLOD(byte[] inData)` constructor